pkg/vmprovider/providers/vsphere/session/session_vm_update.go

```diff
@@ -663,6 +673,42 @@ func (s *Session) prepareVMForPowerOn(
 	return nil
 }
 
+const (
+	// CAPWClusterRoleLabelKey is the key for the label applied to a VM that was
+	// created by CAPW.
+	CAPWClusterRoleLabelKey = "capw.vmware.com/cluster.role"
+
+	// CAPVClusterRoleLabelKey is the key for the label applied to a VM that was
+	// created by CAPV.
+	CAPVClusterRoleLabelKey = "capv.vmware.com/cluster.role"
+)
+
+func PropagateSearchSuffixesForTKGNodes(
+	labels map[string]string,
+	updateArgs *VMUpdateArgs,
+	client ctrl.Client) error {
+
+	// Propagate the DNS search suffixes from the config map if this is a TKGs
+	// node. Please note that this will only be used by guests for VMs created
+	// by TKGs that use Cloud-Init. Guest OS Customization (GOSC) has no means
+	// to set the DNS search suffix.
+	_, ok := labels[CAPWClusterRoleLabelKey]
+	if !ok {
+		_, ok = labels[CAPVClusterRoleLabelKey]
+	}
+	if !ok {
+		return nil
+	}
+
+	searchSuffixes, err := config.GetSearchSuffixesFromConfigMap(client)
+	if err != nil {
```

Comment:
Unless something has changed in the last year, gce2e doesn't create this CM - that's why the `GetNameserversFromConfigMap()` error is swallowed 😞 .  There's a whole lot of cruft left in this code - I think having `GetNameserversFromConfigMap()` (renaming to taste) return this value too would make sense so the CM is fetched in a single place, but all this stuff needs to be rewritten sooner than later anyways.

Reply:
Yeah, I was really torn between keeping to the patterns or refactoring. I would love to decompose this to divorce the testing responsibility a bit. But because I'm doing the logic to check if the label exists in the function, it means I should not read the ConfigMap outside the function. I thought about passing in `func() (*ConfigMap, error)`?

Reply:
Thoughts on this?

Reply:
Also, will refactor so that the other function returns both.